### PR TITLE
DOCSP-43473: oidc

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43473>
Staging - https://deploy-preview-382--docs-golang.netlify.app/fundamentals/enterprise-auth/#mongodb-oidc

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x]  Did you run a grammar-check?
- [ ] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/enterprise-auth.txt [hunk 1]
```diff
@@ -197,16 +197,324 @@ address of your MongoDB server:
...
+
+.. _golang-mongodb-oidc-azure-imds:
+
+Azure IMDS
+~~~~~~~~~~
+
+If your application runs on an Azure VM, or otherwise uses the
+`Azure Instance Metadata Service <https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service>`__
+(IMDS), you can authenticate to MongoDB by using the {+driver-short+}'s
+built-in Azure support.
```
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/fundamentals/enterprise-auth.txt:231]: edit: based on the rest of the PR, looks like you've accounted for that!

**Review by mongoKart — APPROVED** [review]
LGTM! awesome job. how was the process of transferring an existing page to Go?

File: source/includes/authentication/gke-callback.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		}, nil
+	}
+	// end-callback
+
+	// start-credential-callback
+	uri := "mongodb://<hostname>:<port>"
+	props := map[string]string{
+		"ENVIRONMENT":    "gcp",
+		"TOKEN_RESOURCE": "<audience>",
+	}
```
Comment (matthewdale) [inline source/includes/authentication/gke-callback.go:31]: Similar to above, custom callbacks do not use `AuthMechanismProperties`. See https://github.com/mongodb/mongo-go-driver/pull/1809 for the related example fix.

```suggestion
```

File: source/includes/authentication/gke-callback.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	opts := options.Client().ApplyURI(uri)
+	opts.SetAuth(
+		options.Credential{
+			AuthMechanism:           "MONGODB-OIDC",
+			AuthMechanismProperties: props,
```
Comment (matthewdale) [inline source/includes/authentication/gke-callback.go:36]: Similar to above, custom callbacks do not use `AuthMechanismProperties`. See https://github.com/mongodb/mongo-go-driver/pull/1809 for the related example fix.

```suggestion
```

**Review by matthewdale — CHANGES_REQUESTED** [review]

File: source/fundamentals/enterprise-auth.txt [hunk 4]
```diff
@@ -197,16 +197,324 @@ address of your MongoDB server:
...
+      your connection string:
+        
+      .. code-block:: go
+
+	      uri := "mongodb://<hostname>:<port>/?" +
+	      	   "username=<Azure client ID or application ID>" +
+	      	   "&authMechanism=MONGODB-OIDC" +
+	      	   "&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:<percent-encoded audience>"
+      
+	      client, err := mongo.Connect(options.Client().ApplyURI(uri))
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/enterprise-auth.txt:302]: This Pr is against the master branch of the docs, which is tracking for the v2 release. When I backport this ticket to the v1.17 branch, I'll make sure to correct that!

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**